buildenv/jenkins/lineEndingsCheck

```diff
@@ -0,0 +1,96 @@
+    node ('master') {
+        timestamps {
+            git url: SRC_REPO
+            sh "git fetch --tags --progress origin +refs/pull/*:refs/remotes/origin/pr/*"
+            sh "git checkout ${sha1}"
+            sh 'git fetch origin'
+            sh "git diff --diff-filter=ACM --name-only origin/${ghprbTargetBranch} HEAD"
+            FILES = sh (
+                script: "git diff --diff-filter=ACM --name-only origin/${ghprbTargetBranch} HEAD",
+                returnStdout: true
+            ).trim()
+            def FILES_LIST = FILES.split("\\r?\\n")
+            FILES_LIST.each() {
+                println "Checking file: '${it}'"
+                TYPE = sh (
+                    script: "file -b '${it}'",
+                    returnStdout: true
+                ).trim()
+
+                switch (TYPE) {
+                    case ~/empty/:
+                        echo "Empty file: '${it}'"
+                        break
+                    case ~/.*text.*/:
+                        switch (it) {
+                            case ~/.*\.bat|.*\.cmd/:
+                                switch(TYPE) {
+                                    case ~/.*CRLF line terminators.*/:
+                                        echo "Good windows script: '${it}' type: '${TYPE}'"
+                                        break
+                                    default:
+                                        echo "ERROR - should have CRLF line terminators: '${it}' type: '${TYPE}'"
+                                        FAIL = true
+                                        BAD_FILES << new String("${it}")
+                                }
+                            default:
+                                switch (TYPE) {
+                                    case ~/.*CRLF line terminators.*/:
+                                        echo "ERROR - should have LF line terminators: '${it}' type: '${TYPE}'"
+                                        FAIL = true
+                                        BAD_FILES << new String("${it}")
+                                        break
+                                    default:
+                                        echo "Good text file: '${it}' type: '${TYPE}'"
+                                        break
+                                }
+                        }
+                    default:
+                        echo "Non-text file: '${it}' type: '${TYPE}'"
+                }
+            }
+            if (FAIL) {
+                echo "${HASHES}"
+                echo "${HASH}"
+                echo "${HASH} The following files were modified and have incorrect line endings"
+                BAD_FILES.each() {
+                    echo "# ${it}"
+                }
+                echo "${HASH}"
+                echo "${HASHES}"
+                sh 'exit 1'
+            } else {
+                echo "All modified files appear to have correct line endings"
+            }
+        }
+    }
+}
```

Comment:
I almost want to suggest that we should reject files without a final line-end (like this!).

Reply:
Not sure what you mean here? Are you suggesting I add an empty line after L96?

Reply:
I meant you should finish line 96 (by adding a newline).